node/network/approval-distribution/src/lib.rs

```diff
@@ -1820,50 +2098,131 @@ pub const MAX_APPROVAL_BATCH_SIZE: usize = ensure_size_not_zero(
 	MAX_NOTIFICATION_SIZE as usize / std::mem::size_of::<IndirectSignedApprovalVote>() / 3,
 );
 
+// Low level helper for sending assignments.
+async fn send_assignments_batched_inner(
+	sender: &mut impl overseer::ApprovalDistributionSenderTrait,
+	batch: Vec<(IndirectAssignmentCertV2, CandidateBitfield)>,
```

Comment:
we can save allocations by having this be an `impl IntoIterator` and cloning the inner values lazily at the call-site

Reply:
done